rmw_connext_shared_cpp/src/node_names.cpp

```diff
@@ -125,26 +136,63 @@ get_node_names(
       RMW_SET_ERROR_MSG("could not allocate memory for node namespace");
       goto fail;
     }
+
+    ++named_nodes_num;
+  }
+
+  // prepare actual output without empty names
+  // TODO: implement functionality that allocates just enough memory for the output buffer,
+  // so using additional buffer would be unnessecary 
+  rcutils_ret = rcutils_string_array_init(node_names, named_nodes_num, &allocator);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+  }
+
+  for (auto i = 0; i < named_nodes_num; ++i) {
+    node_names->data[i] = rcutils_strdup(tmp_names_list.data[i], allocator);
+    tmp_names_list.data[i] = nullptr;
   }
 
+  // prepare actual output without empty names
+  // TODO: implement functionality that allocates just enough memory for the output buffer,
+  // so using additional buffer would be unnessecary 
+  rcutils_ret = rcutils_string_array_init(node_names, named_nodes_num, &allocator);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+  }
+
+  for (auto i = 0; i < named_nodes_num; ++i) {
+    node_names->data[i] = rcutils_strdup(tmp_names_list.data[i], allocator);
+    tmp_names_list.data[i] = nullptr;
+  }
+
+  rcutils_ret = rcutils_string_array_fini(&tmp_names_list);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
+  }
   return RMW_RET_OK;
+
 fail:
   if (node_names) {
     rcutils_ret = rcutils_string_array_fini(node_names);
     if (rcutils_ret != RCUTILS_RET_OK) {
-      RCUTILS_LOG_ERROR_NAMED(
-        "rmw_connext_cpp",
-        "failed to cleanup during error handling: %s", rcutils_get_error_string().str);
-      rcutils_reset_error();
+      RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+      return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
     }
   }
+  rcutils_ret = rcutils_string_array_fini(&tmp_names_list);
+  if (rcutils_ret != RCUTILS_RET_OK) {
+    RMW_SET_ERROR_MSG(rcutils_get_error_string().str);
+    return rmw_convert_rcutils_ret_to_rmw_ret(rcutils_ret);
```

Comment:
Instead of setting an error message and returning, log an error and reset the error:

```c++
RCUTILS_LOG_ERROR_NAMED(
  "rmw_connext_cpp",
  "failed to cleanup during error handling: %s", rcutils_get_error_string().str);
rcutils_reset_error();
```

Reply:
Done